cmd/siac/main.go

```diff
@@ -118,5 +242,133 @@ func main() {
 		} else {
 			fmt.Printf("%s SC\n", balance.Siacoins)
 		}
+
+	case walletAddressCmd:
+		if len(args) != 0 {
+			cmd.Usage()
+			return
+		}
+		address, err := getClient().WalletAddress()
+		check("couldn't get address:", err)
+		fmt.Println(address.Address)
+
+	case walletAddressesCmd:
+		if len(args) != 0 {
+			cmd.Usage()
+			return
+		}
+		addresses, err := getClient().WalletAddresses()
+		check("couldn't get address:", err)
+		for _, address := range addresses.Addresses {
+			fmt.Println(address)
+		}
+
+	case walletTransactionsCmd:
+		if len(args) != 0 {
+			cmd.Usage()
+			return
+		}
+		transactions, err := getClient().WalletTransactions()
+		check("couldn't get transactions:", err)
+
+		w := makeTabWriter()
+		defer w.Flush()
+
+		fmt.Fprintf(w, "%v\t%v\t%v\t%v\n", "Index", "ID", "Inflow", "Outflow")
+		for _, txn := range transactions.Transactions {
+			fmt.Fprintf(w, "%v\t%v\t%v\t%v\n", txn.Index, txn.ID, txn.Inflow, txn.Outflow)
+		}
+
+	case walletSignCmd:
+		if len(args) == 0 {
+			cmd.Usage()
+			return
+		}
+
+		var txn types.Transaction
+		if err := json.Unmarshal([]byte(args[0]), &txn); err != nil {
+			// if we can't decode it as json, try to see if it is a file
+			file, err := os.Open(args[0])
+			check("failed to open JSON file:", err)
+			check("failed to decode JSON:", json.NewDecoder(file).Decode(&txn))
+		}
+		var toSign []types.ElementID
+		if len(args) > 1 {
+			if err := json.Unmarshal([]byte(args[1]), &toSign); err != nil {
+				// if we can't decode it as json, try to see if it is a file
+				file, err := os.Open(args[1])
+				check("failed to open JSON file:", err)
+				check("failed to decode JSON:", json.NewDecoder(file).Decode(&toSign))
+			}
+		}
+		signed, err := getClient().WalletSign(txn, toSign)
+		check("failed to sign transaction:", err)
+		encoded, err := json.Marshal(signed.Transaction)
+		check("failed to marshal signed transaction:", err)
+		fmt.Println(string(encoded))
+
+	case txpoolCmd:
+		cmd.Usage()
+
+	case txpoolBroadcastCmd:
+		if len(args) != 1 {
+			cmd.Usage()
+			return
+		}
+
+		var txn types.Transaction
+		if err := json.Unmarshal([]byte(args[0]), &txn); err != nil {
```

Comment:
on further thought, it's unlikely that anyone will be passing a big JSON object as a CLI argument. Let's just require files everywhere.  You can copy the `readTxn` and `writeTxn` helpers from `walrus-cli`: https://github.com/lukechampine/walrus-cli/blob/0f2399a1163f695082275b83aff9974385c71739/main.go#L147-L161